mail-filter/spamassassin/spamassassin-3.4.4-r1.ebuild

```diff
@@ -0,0 +1,321 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit perl-functions systemd toolchain-funcs eapi7-ver
+
+MY_P="Mail-SpamAssassin-${PV//_/-}"
+S="${WORKDIR}/${MY_P}"
+DESCRIPTION="An extensible mail filter which can identify and tag spam"
+HOMEPAGE="https://spamassassin.apache.org/"
+SRC_URI="mirror://apache/spamassassin/source/${MY_P}.tar.bz2"
+
+LICENSE="Apache-2.0 GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-linux ~x86-linux ~x86-macos"
+IUSE="berkdb cron ipv6 ldap libressl mysql postgres qmail sqlite ssl test"
+RESTRICT="!test? ( test )"
+
+# The Makefile.PL script checks for dependencies, but only fails if a
+# required (i.e. not optional) dependency is missing. We therefore
+# require most of the optional modules only at runtime.
+REQDEPEND="acct-user/spamd
+	acct-group/spamd
+	dev-lang/perl:=
+	dev-perl/HTML-Parser
+	dev-perl/Net-DNS
+	dev-perl/NetAddr-IP
+	virtual/perl-Archive-Tar
+	virtual/perl-Digest-SHA
+	virtual/perl-IO-Zlib
+	virtual/perl-Time-HiRes
+	ssl? (
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl )
+	)"
+
+# SpamAssassin doesn't use libwww-perl except as a fallback for when
+# curl/wget are missing, so we depend on one of those instead. Some
+# mirrors use https, so we need those utilities to support SSL.
+#
+# re2c is needed to compile the rules (sa-compile).
+#
+# We still need the old Digest-SHA1 because razor2 has not been ported
+# to Digest-SHA.
+OPTDEPEND="app-crypt/gnupg
+	dev-perl/BSD-Resource
+	dev-perl/Digest-SHA1
+	dev-perl/Encode-Detect
+	dev-perl/Geo-IP
+	dev-perl/HTTP-Date
+	dev-perl/Mail-DKIM
+	dev-perl/Mail-SPF
+	dev-perl/Net-Patricia
+	dev-perl/Net-CIDR-Lite
+	dev-util/re2c
+	|| ( net-misc/wget[ssl] net-misc/curl[ssl] )
+	virtual/perl-MIME-Base64
+	virtual/perl-Pod-Parser
+	berkdb? ( virtual/perl-DB_File )
+	ipv6? ( dev-perl/IO-Socket-INET6 )
+	ldap? ( dev-perl/perl-ldap )
+	mysql? (
+		dev-perl/DBI
+		dev-perl/DBD-mysql
+	)
+	postgres? (
+		dev-perl/DBI
+		dev-perl/DBD-Pg
+	)
+	sqlite? (
```

Comment:
I'm not sure I follow.  Enabling the `sqlite` use flag will install the dependencies required by that database so it will function with that backend.

Reply:
Because `+sqlite` or `-sqlite` won't make any difference of the compiled product, it is useless to compile this program again whether that flag changes. It's banned by QA, see 
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001
(unless the build system has some automatic detection whether it's available or not - then you need to patch it to make it conditional)

Reply:
Well, being perl, it may have some sort of runtime detection, or alternatively, a path users can sometimes use on the installed code, but without the dependency, will have a fatal error.

Thus, depending on how often its needed, you'll want it in RDEPEND for some userbase.

But some users will not want the added weight and complexity the dependency produces, which in the case of DBD-SQLite, involves compiling a copy of SQLite ( because it can't use the system sqlite reliably )

This is really a place where there are no good options currently.

Based on the user needing to install the right SA to work, it seems like this situation in perl could fall under "such as a package being nonfunctional unless at least one of a set of optional runtime dependencies is installed."

Reply:
This setup has existed since at least 2006.  (I see it in the 3.1.1 ebuild, it probably pre-dates that even.)  Changing it may break a lot of people's setups.  I think in this case it makes more sense to leave it as it is rather than expect users to have to manually manage a ton of dependencies just because Perl happens to be an interpreted language rather than a compiled one.

Reply:
Okay, thanks for explaining.